modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/data-engine/DataEngineLayoutBuilderHandler.es.js

```diff
@@ -88,6 +88,11 @@ export default function DataEngineLayoutBuilderHandler({namespace}) {
 		const editingLanguageId = event.item.getAttribute('data-value');
 		const dataLayoutBuilder = await getDataLayoutBuilder();
 
+		dataLayoutBuilder.current.dispatch({
```

Comment:
The `EVENT_TYPES.LANGUAGE.ADD` is always preceded by an `EVENT_TYPES.LANGUAGE.CHANGE`, so to avoid successive dispatches (that could affect app performance by unescessay component rendering) the `LANGUAGE.CHANGE` code could be extracted to a function and reused into  `LANGUAGE.ADD` code on `/modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/reducers/languageReducer.es.js`

Note: Remember to also change in on `/modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/new-js/components/field-sets/FieldSetModal.js` and `/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/components/TranslationManager.es.js`

Reply:
Sure thing, I did it here https://github.com/liferay-forms/liferay-portal/pull/245/commits/59fd3347bd8c3d93ed2450bb87d62a905938d398. Could you take a look on it? @miltonmc